crates/interfaces/src/p2p/full_block.rs

```diff
@@ -197,6 +228,60 @@ enum ResponseResult {
     Body(PeerRequestResult<Option<BlockBody>>),
 }
 
+/// The response of a body request.
+#[derive(Debug)]
+enum BodyResponse {
+    /// Already validated against transaction root of header
+    Validated(BlockBody),
+    /// Still needs to be validated against header
+    PendingValidation(WithPeerId<BlockBody>),
+}
+
+/// Ensures the block response data matches the header.
+///
+/// This ensures the body response items match the header's hashes:
+///   - ommer hash
+///   - transaction root
+///   - withdrawals root
+fn ensure_valid_body_response(
+    header: &SealedHeader,
+    block: &BlockBody,
+) -> Result<(), ConsensusError> {
```

Comment:
we already have `validate_block_standalone` fn

Reply:
this requires chain spec, which we don't have here

Reply:
could integrate this in a followup but requires a few changes